plugin/server_audit/server_audit.c

```diff
@@ -1401,6 +1403,14 @@ static size_t log_header(char *message, size_t message_len,
     host_len= userip_len;
     host= userip;
   }
+  char port_str[16];
+  if (port == NULL) {
+    strncpy(port_str, "unavailable", sizeof(port_str) - 1);
+    port_str[sizeof(port_str) - 1] = '\0'; 
+  } else {
+    snprintf(port_str, sizeof(port_str) - 1, "%u", port); // Subtract 1 to account for null terminator
+    port_str[sizeof(port_str) - 1] = '\0'; // Ensure null termination
```

Comment:
Since the maximum possible length is 10 and you have allocated 16, this shouldn't be needed. Also `my_snprintf` would do this for you anyway.

Reply:
Removed the null termination and using my_snprintf now. Thank you.